app/src/main/java/fi/helsinki/opintoni/service/converter/PortfolioConverter.java

```diff
@@ -56,12 +62,47 @@ public PortfolioDto toDto(Portfolio portfolio) {
         portfolioDto.visibility = portfolio.visibility;
         portfolioDto.avatarUrl = avatarImageService.getPortfolioAvatarImageUrl(portfolio.getOwnerId());
         portfolioDto.componentVisibilities = componentVisibilityService.findByPortfolioId(portfolio.id);
+
+        fetchPortfolioComponents(portfolio.id, portfolioDto, componentFetchStrategy);
+
         return portfolioDto;
     }
 
+    private void fetchPortfolioComponents(Long portfolioId,
+                                          PortfolioDto portfolioDto,
+                                          ComponentFetchStrategy componentFetchStrategy) {
+        if(componentFetchStrategy == ComponentFetchStrategy.ALL) {
+            fetchAllComponents(portfolioId, portfolioDto);
+        } else if(componentFetchStrategy == ComponentFetchStrategy.PUBLIC) {
+            fetchPublicComponents(portfolioId, portfolioDto);
+        }
```

Comment:
Se oli ensin switch-case, mut koska siitä tuli paljon noisea `break`it, niin muutin `if`fiksi